kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala

```diff
@@ -0,0 +1,236 @@
+  useDefaultPropertyValueIfMissing()
+
+  validateArguments()
+
+  private def useDefaultPropertyValueIfMissing(): Unit = {
+    if (zkAddress == null) {
+      zkAddress = defaultKyuubiProperties.getOrElse(HA_ZK_QUORUM.key, null)
+    }
+    if (nameSpace == null) {
+      nameSpace = defaultKyuubiProperties.getOrElse(HA_ZK_NAMESPACE.key, null)
+    }
+    if (version == null) {
+      version = KYUUBI_VERSION
+    }
+  }
+  /** Ensure that required fields exists. Call this only once all defaults are loaded. */
+  private def validateArguments(): Unit = {
+    action match {
+      case KyuubiCtlAction.CREATE => validateActionArguments()
+      case KyuubiCtlAction.GET => validateActionArguments()
+      case KyuubiCtlAction.DELETE => validateActionArguments()
+      case KyuubiCtlAction.LIST => validateActionArguments()
+    }
+  }
+
+  private def validateActionArguments(): Unit = {
+    if (zkAddress == null) {
+      fail("Zookeeper address is not specified and no default value to load")
+    }
+    if (nameSpace == null) {
+      fail("Zookeeper namespace is not specified and no default value to load")
+    }
+    if (version == null) {
+      fail("Kyuubi version is not specified and could not found KYUUBI_VERSION in " +
+        "kyuubi-build-info")
+    }
+    if (host == null) {
+      fail("Must specify host for service")
+    }
+    if (port == null) {
+      fail("Must specify port for service")
+    }
+    if (role == KyuubiCtlActionRole.ENGINE && user == null) {
+      fail("Must specify user name for engine")
+    }
+  }
+
+  private def printUsageAndExit(exitCode: Int, unknownParam: Any = null): Unit = {
+    if (unknownParam != null) {
+      info("Unknown/unsupported param " + unknownParam)
+    }
+    val command = sys.env.getOrElse("_KYUUBI_CMD_USAGE",
+      """Usage: kyuubi-ctl <create|get|delete|list>  <server|engine> --zkAddress ...
+        |--namespace ... --user ... --host ... --port ... --version""".stripMargin)
+    info(command)
+
+    info(
+      s"""
+         |Operations:
+         |  - create                    expose a service to a namespace, this case is rare but
+         |                              sometimes we may want one server to be reached in 2 or more
+         |                              namespaces by different user groups
+         |  - get                       get the service node info
+         |  - delete                    delete the specified serviceNode
+         |  - list                      list all the service nodes for a particular domain
+         |
+         |Role:
+         |  - server                    default
+         |  - engine
+         |
+         |Args:
+         |  --zkAddress                 one of the zk ensemble address, using kyuubi-defaults/conf if absent
+         |  --namespace                 the namespace, using kyuubi-defaults/conf if absent
+         |  --user
+         |  --host
+         |  --port
+         |  --version
```

Comment:
--version      using the compiled KYUUBI_VERSION default, change it if the active service is running in another